### PR TITLE
tracing: add ExtractTraceSpanID

### PR DESCRIPTION
**Pull request by krajorama**
**What this PR does**:

Add ExtractTraceSpanID helper function for enhanced logging.

**Which issue(s) this PR fixes**:

Related to https://github.com/grafana/mimir/pull/8599

**Checklist**
- [ ] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by flxbk — APPROVED** [review]
LGTM

File: tracing/tracing.go [hunk 1]
```diff
@@ -55,16 +55,30 @@ func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, err
...
+		return nil
 	}
 	sctx, ok := sp.Context().(jaeger.SpanContext)
 	if !ok {
-		return "", false
+		return nil
 	}
-
-	return sctx.TraceID().String(), true
+	return &sctx
```
Comment (pstibrany) [inline tracing/tracing.go:81]: By returning `&sctx`, `jaeger.SpanContext` needs to be allocated on a heap first, and then pointer to it returned, right?

Perhaps we could return just traceID and spanID directly? Or `jaeger.SpanContext` without the pointer.

**Review by pstibrany — APPROVED** [review]

File: tracing/tracing.go (hunk 1 above)
Comment (krajorama) [reply tracing/tracing.go:81]: I thought about that , but that would be duplicated code. I can definitely do a copy of the jeager.SpanContext thought, if that's preferred?

Comment (krajorama) [reply tracing/tracing.go:81]: By the way , I don't know if the compiler is smart enough to avoid allocation on the heap

Comment (ying-jeanne) [reply tracing/tracing.go:81]: Can we keep ExtractTraceID, and call ExtractTraceSpanID inside, just discard spanID when return result. In that case no need to refectory this extractJaegerContext but leave all logics in ExtractTraceSpanID

Comment (krajorama) [reply tracing/tracing.go]: facepalm, of course

**Review by ying-jeanne — APPROVED** [review]
LGTM

File: tracing/tracing.go (hunk 1 above)
Comment (ying-jeanne) [reply tracing/tracing.go]: I think it is a fair comment, current code LGTM, approved.

Comment (krajorama) [reply tracing/tracing.go]: it did inspire me to go ahead and look at the assembly :) Got the final result by deferring the formatting to the public function , but only returning the two relevant fields from the helper function. It's the least extra instructions.

**Pull request merged**